File: sdk/go/common/util/cmdutil/testdata/term_frozen.py
import signal
import sys
import time

def signal_handler(signal, frame):
    time.sleep(3)
    print("error: was not forced to exit", file=sys.stderr)
    sys.exit(2)

signal.signal(signal.SIGINT, signal_handler)
if hasattr(signal, "SIGBREAK"):
    # SIGBREAK is only available on Windows
    signal.signal(signal.SIGBREAK, signal_handler)
print("ready", flush=True)

time.sleep(3)
print("error: signal not received", file=sys.stderr)
sys.exit(1)

File: sdk/go/common/util/cmdutil/testdata/term_frozen.go
//go:build ignore
// +build ignore

package main

import (
	"fmt"
	"os"
	"os/signal"
	"time"
)

func main() {
	sigch := make(chan os.Signal, 1)
	signal.Notify(sigch, os.Interrupt)
	// os.Interrupt handles SIGINT and CTRL_BREAK_EVENT.

	fmt.Println("ready")
	select {
	case <-sigch:
		time.Sleep(3 * time.Second)
		fmt.Fprintln(os.Stderr, "error: was not forced to exit")
		os.Exit(2)

	case <-time.After(3 * time.Second):
		fmt.Fprintln(os.Stderr, "error: did not receive signal")
		os.Exit(1)
	}
}
